### ISSUE TITLE
Challenges: The approach of teaching web development is not accessibility centered.

### ISSUE THREAD

**Issue by HarryAdney** [w3c/wcag#1173]
I've read, heard and watched a lot of tutorials as well as taken commercial courses on web development over some 13-odd years of involvement in the web-based industry and never was accessibility addressed. Not once.

It's been the elephant in the room that experienced devs and designers have been aware of, but nobody has wanted to address because 1, noboby else was and 2, it was perceived as being hard to do.

Maybe more awareness and fines should be introduced. In the meantime, the various IDEs, frameworks etc. which detect invalid markup as part of their process (I'm referring to things like missing end tags, empty alt, etc.) could introduce a similar kind of "mode" - maybe, to highlight WCAG errors.

I think web devs, tutors, and framework developers need to move to an "accessibility first" mindset just like we moved to a mobile first mindset under web 2.0.

**Comment by mgifford** [comment]
I am hopeful for the GAAD Pledge - https://diamond.la/GAADPledge - it would be great to see React be a leader in this space.

I've been really impressed with https://www.gatsbyjs.org

We've been trying to build this mindset into the https://www.drupal.org community too, with some success:
https://webaim.org/projects/million/#CMS

The mindset is that accessible code is good code. Accessibility barriers are bugs not features.
